clients/fides-js/src/lib/tcf/vendors.ts

```diff
@@ -10,19 +17,65 @@ export const vendorIsGvl = (
   return gvl.vendors[vendor.id];
 };
 
-export const vendorRecordsWithLegalBasis = (
-  records: TCFVendorRecord[],
-  legalBasis: LegalBasisForProcessingEnum
-) =>
-  records.filter((record) => {
-    const { purposes, special_purposes: specialPurposes } = record;
-    const hasApplicablePurposes = purposes?.filter((purpose) =>
-      purpose.legal_bases?.includes(legalBasis)
-    );
-    const hasApplicableSpecialPurposes = specialPurposes?.filter((purpose) =>
-      purpose.legal_bases?.includes(legalBasis)
-    );
-    return (
-      hasApplicablePurposes?.length || hasApplicableSpecialPurposes?.length
-    );
+const transformVendorDataToVendorRecords = ({
```

Comment:
this function is one of the larger changes in this PR. basically the vendor tab wants both vendors and systems together. on top of that, it also wants it combined across legal basis. this function returns a type that makes that easy